components/dada-ir/src/code.rs

```diff
@@ -14,6 +14,12 @@ pub struct Code {
     /// Tokens for the parameter list (parsed when we generate the syntax tree).
     pub parameter_tokens: Option<TokenTree>,
 
+    /// Span of `->` symbol if it has one, or where it would go if doesn't have it.
+    pub return_type: FileSpan,
```

Comment:
See my other comment about why I try to avoid including a span directly in this structure -- in short, if the only thing that has changed is the span, I don't want to have to recompile.

Reply:
I have created `ReturnType` entity and also added a `ReturnTypeKind` enum to represent unit or value.